src/pyscheduling/Problem.py

```diff
@@ -4,14 +4,19 @@
 from enum import Enum
 from pathlib import Path
 from time import perf_counter
+import numpy as np
+import random
 
 Job = namedtuple('Job', ['id', 'start_time', 'end_time'])
 
+class GenerationProtocol(Enum):
+    BASE = 1
+
 class GenerationLaw(Enum):
     UNIFORM = 1
     NORMAL = 2
 
-class Constraints(Enum):
+class Constraints_Enum(Enum):
```

Comment:
It's better to name this ConstraintsEnum to respect PEP8

"Class names should normally use the CapWords convention"